react-validated-input-component/my-app/src/ValidatedInput.js

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+
+export default function ValidatedInput() {
+  const [pass, setPass] = useState('');
+  const [invalid, setInvalid] = useState(true)
+  const [text, setText] = useState('Password is required.')
+  const [icon, setIcon] = useState('')
+  function handleSubmit(e) {
+    e.preventDefault();
+    console.log('Controlled', { pass } )
+  }
+
+  function handlePassword(e)  {
+    setPass(e.target.value.trim().replaceAll(' ', ''));
+    setIcon('❌')
+    if (e.target.value.length === 0) {
+      setText('Password is required.')
+      setIcon('❌')
+    }
+    if (e.target.value.length < 8 && e.target.value.length > 0) {
+    setText('Password must be 8 Characters long')
+    } else if (e.target.value.length >= 8) {
+    setText('')
+    setIcon('✅')
+  }
+  }
+
+  return (
+    <form id="login-form" onSubmit={(e) => handleSubmit(e)}>
```

Comment:
This exercise doesn't need a form. It should do all the validation as you type. The intent is for this component to be usable in any form. By wrapping it in a form element, you make it so it cannot be used anywhere else since forms cannot be nested.